Content.Shared/Damage/Systems/DamageableSystem.cs

```diff
@@ -256,6 +256,39 @@ public void SetAllDamage(EntityUid uid, DamageableComponent component, FixedPoin
             // Shitmed Change End
         }
 
+        /// <summary>
+        ///     Changes all damage types supported by a <see cref="DamageableComponent"/> by the specified value.
+        /// </summary>
+        /// <remakrs>
+        ///     Will not lower damage to a negative value.
+        /// </remakrs>
+        public void ChangeAllDamage(EntityUid uid, DamageableComponent component, FixedPoint2 addedValue)
+        {
+            foreach (var type in component.Damage.DamageDict.Keys)
+            {
+                component.Damage.DamageDict[type] += addedValue;
+                if (component.Damage.DamageDict[type] < 0)
+                    component.Damage.DamageDict[type] = 0;
+            }
+
+            // Changing damage does not count as 'dealing' damage, even if it is set to a larger value, so we pass an
+            // empty damage delta.
+            DamageChanged(uid, component, new DamageSpecifier());
+
+            // Shitmed Change Start
+            if (HasComp<TargetingComponent>(uid))
+            {
+                foreach (var (part, _) in _body.GetBodyChildren(uid))
+                {
+                    if (!TryComp(part, out DamageableComponent? damageComp))
+                        continue;
+
+                    ChangeAllDamage(part, damageComp, addedValue);
+                }
+            }
+            // Shitmed Change End
+        }
```

Comment:
you can make this a guard clause

Reply:
I'm not familiar with the term-- do you mean this?
```suggestion
            // Shitmed Change Start
            if (!HasComp<TargetingComponent>(uid))
                continue;
                
            foreach (var (part, _) in _body.GetBodyChildren(uid))
            {
                if (!TryComp(part, out DamageableComponent? damageComp))
                    continue;

                ChangeAllDamage(part, damageComp, addedValue);
            }
            // Shitmed Change End
        }
```